include/circt-c/Dialect/FIRRTL.h

```diff
@@ -17,8 +17,94 @@
 extern "C" {
 #endif
 
+/// Wraps a FIRRTL BundleType element.
+struct FirrtlBundleElement {
+  MlirIdentifier name;
+  bool isFlip;
+  MlirType type;
+};
+typedef struct FirrtlBundleElement FirrtlBundleElement;
+
+//===----------------------------------------------------------------------===//
+// Dialect API.
+//===----------------------------------------------------------------------===//
+
 MLIR_DECLARE_CAPI_DIALECT_REGISTRATION(FIRRTL, firrtl);
 
+//===----------------------------------------------------------------------===//
+// Type API.
+//===----------------------------------------------------------------------===//
+
+/// Return 'true' if this is any FIRRTL type.
+bool firrtlTypeIs(MlirType type);
+
+/// Return 'true' if this is a FIRRTL ground type.
+bool firrtlTypeIsGround(MlirType type);
+
+/// Return 'true' if this is a FIRRTL const type.
+bool firrtlTypeIsConst(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Clock type.
+bool firrtlTypeIsClock(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Reset type.
+bool firrtlTypeIsReset(MlirType type);
+
+/// Return 'true' if this is a FIRRTL AsyncReset type.
+bool firrtlTypeIsAsyncReset(MlirType type);
+
+/// Return 'true' if this is a FIRRTL SInt type.
+bool firrtlTypeIsSInt(MlirType type);
+
+/// Return 'true' if this is a FIRRTL UInt type.
+bool firrtlTypeIsUInt(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Analog type.
+bool firrtlTypeIsAnalog(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Bundle type.
+bool firrtlTypeIsBundle(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Vector type.
+bool firrtlTypeIsFVector(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Reference type.
+bool firrtlTypeIsRef(MlirType type);
+
+// bool firrtlTypeIsOpenBundle(MlirType type);
```

Comment:
Is there any problem with these? They seems to be implemented. I'd suggest to either uncomment this or remove it either way.